src/components/Dialogs/FileBrowser/FileBrowserDialogComponent.tsx

```diff
@@ -20,18 +20,47 @@ export class FileBrowserDialogComponent extends React.Component {
     @observable debouncedFilterString: string = "";
     @observable defaultWidth: number;
     @observable defaultHeight: number;
+    @observable enableImageArithmetic: boolean = false;
+    @observable imageArithmeticString: string = "";
+    private readonly imageArithmeticInputRef: React.RefObject<HTMLInputElement>;
 
     constructor(props: any) {
         super(props);
         makeObservable(this);
         this.defaultWidth = 1200;
         this.defaultHeight = 600;
+        this.imageArithmeticInputRef = React.createRef<HTMLInputElement>();
     }
 
     private handleTabChange = (newId: TabId) => {
         FileBrowserStore.Instance.setSelectedTab(newId);
     };
 
+    @action private handleFileClicked = (file: ISelectedFile) => {
+        FileBrowserStore.Instance.selectFile(file);
+        if (this.enableImageArithmetic) {
+            // Check if the existing string has a trailing quote or not
+            const quoteRegex = /(["'])+/gm;
+            const quoteCount = this.imageArithmeticString.match(quoteRegex)?.length;
+            const trailingQuote = quoteCount % 2 !== 0;
+
+            const operatorRegex = /(\+|-|\*|\/\()$/gm;
```

Comment:
is there a `|` missing between `\/` and `\(`?
shall we also handle spaces in the end, e.g. `"filename"  +  `?

Reply:
adjusted regex to allow for optional whitespace after the operator. Also added in `,` to the list, so that it works for situations like `atan2("file1.fits", `